File: Chapter-05-Linked-Lists/SList-Front/SList-Front.py
# Return the value at the head of the list. return null if list is empty

# import the list class
import sys
from os.path import abspath, dirname

mainDir = dirname(dirname(abspath(__file__)))
sys.path.append(mainDir)

from singleList import List

# import the random module
from random import randrange

# create a new list and populate it
newSList = List()
for i in range(0, 10):
    newSList.add(randrange(100))

# function to return the first value
def front(head):
    if not head:
        return None
    else:
        return head.val

# print the list
print('The list is {}').format(newSList.show())

# print the value at the head of the list
print('The value at the head of the list is {}').format(front(newSList.getHeadNode()))

File: Chapter-05-Linked-Lists/SList-Front/SList-Front.js
// Return the value at the head of the list. return null if list is empty

// import the list class
var path = require('path'),
    SList= require(path.join(__dirname, '../singleList'))

// create a new list and populate it
var newSList = new SList()
for(let i = 0; i < 10; i++){
  newSList.add(Math.floor(Math.random() * 100))
}

// function to return the first value
function front(head){
  if (!head) {
    return null
  } else {
    return head.val
  }
}

// print the list
console.log(`The list is ${newSList.show()}`)

// print the value at head of the list
console.log(`The value at the head of the list is ${front(newSList.getHeadNode())}`)
